### PR TITLE
Scala Common Enrich: 2.11 related changes

### PR DESCRIPTION
**Pull request by BenFradet**

### CODE REVIEW

**Comment by BenFradet** [comment]
I still have to split a couple of commits for #3007 and #3008 as I just saw them.

**Comment by BenFradet** [comment]
Depends on snowplow/scala-forex#110

File: 3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/eventEnrichmentSpecs.scala [hunk 1]
```diff
@@ -67,7 +67,7 @@ class ExtractEventTypeSpec extends Specification with DataTables with Validation
   def e3 =
     "SPEC NAME"          || "INPUT VAL"     | "EXPECTED OUTPUT"                                                                                    |
     "None"               !! None            ! "No collector_tstamp set".fail                                                                       |
-    "Negative timestamp" !! BCTstamp        ! "Collector timestamp -0030-01-01 00:00:00.000 is negative and will fail the Redshift load".fail |
+    "Negative timestamp" !! BCTstamp        ! "Collector timestamp -0030-01-01 00:50:39.000 is negative and will fail the Redshift load".fail |
```
Comment (BenFradet) [inline 3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/eventEnrichmentSpecs.scala]: Haven't found out why this is happening.

File: 3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/sqlquery/OutputSpec.scala [hunk 2]
```diff
@@ -42,6 +42,6 @@ class OutputSpec extends Specification with ValidationMatchers { def is =
 
   def e4 = {
     val date = new Date(1465558727000L)
-    JsonOutput.getValue(date, "java.sql.Date") must beEqualTo(JString("2016-06-10T11:38:47.000Z"))
+    JsonOutput.getValue(date, "java.sql.Date") must beEqualTo(JString("2016-06-10T13:38:47.000+02:00"))
```
Comment (BenFradet) [inline 3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/sqlquery/OutputSpec.scala]: Same here, given that the json4s version didn't change I don't why the default date formats would have changed.

Comment (chuwy) [inline 3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/sqlquery/OutputSpec.scala]: What is your `java -version` output @BenFradet? I noticed differences in output when compiled with java 8.

Comment (BenFradet) [inline 3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/sqlquery/OutputSpec.scala]: Might be that as I'm on java 8

Comment (chuwy) [inline 3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/sqlquery/OutputSpec.scala]: I'd recommend to use our vagrant setup to make sure that our environments are identical.

**Comment by BenFradet** [comment]
@chuwy how would we go about publishing this to central?

**Comment by chuwy** [comment]
Hey @BenFradet,

1. I just finished review. This looks absolutely perfect, I have no notes.
2. Main important moment for CI/CD is that we need to build a DAG of dependencies in case when both lib (SCE) and app (for example, spark enrich) are included in release. Our algorithm looks like following: for each DAG leafs (spark enrich, spark shred) make `publishLocal` for all its dependencies. Iff app (spark enrich) tests are fine we need to publish lib (SCE) to central and then publish app (spark enrich). This is approach taken for snowplow/iglu (another multi-repo). I think above algorithm should work with our release manager (`build_commands`).

/cc @ninjabear @alexanderdean

**Comment by BenFradet** [comment]
closed in favor of #3203

**Pull request closed without merging**